### PR TITLE
feat(last):add last 

### LINKED ISSUE #136: Support for `last`
Hello, @raon0211 
Can I work `last` of #91

### PR DESCRIPTION
**Pull request by minchodang**
close #136. 

### description
I have created a function that returns the last element of an array. Below is a performance comparison with Lodash. If you have any questions or comments, please feel free to share.

Thank you.

### benchmark
스크린샷 2024-07-09 오후 4 59 45

### CODE REVIEW

File: docs/ko/reference/array/last.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+const arr2: number[] = [];
+const result = last(arr2);
+// result는 undefined에요.
+
+const largeArray = Array(1000)
+  .fill(0)
+  .map((_, i) => i);
+const result = last(largeArray);
+// result는 99에요.
```
Comment (jgjgill) [inline docs/ko/reference/array/last.md:41]: I think `999` 🧐

Comment (minchodang) [reply docs/ko/reference/array/last.md:41]: oh! i mistake. thank you comment!

File: docs/reference/array/last.md
[the remark this answers is not in the document]
Comment (minchodang) [reply docs/reference/array/last.md:1]: @jgjgill  Oh thank you. 
I'll add that and push it out as well.

File: docs/ko/reference/array/last.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+# last
+
+배열의 마지막 요소를 반환해요.
+
+이 함수는 배열을 입력 받아 해당 배열의 첫 번 째 요소를 반환해요. 빈 배열의 경우, `undefined`를 반환해요.
```
Comment (raon0211) [inline docs/ko/reference/array/last.md:5]:
```suggestion
빈 배열의 경우, `undefined`를 반환해요.
```

File: docs/ko/reference/array/last.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+## 인터페이스
+
+```typescript
+export function last<T>(arr: readonly [...T[], T]): T;
+export function last<T>(arr: readonly T[]): T | undefined;
```
Comment (raon0211) [inline docs/ko/reference/array/last.md:11]:
```suggestion
function last<T>(arr: [...T[], T]): T;
function last<T>(arr: T[]): T | undefined;
```

File: src/array/last.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+ * const noElement = last(emptyArr);
+ * // noElement will be undefined
+ */
+
+export function last<T>(arr: readonly [...T[], T]): T;
+export function last<T>(arr: readonly T[]): T | undefined;
+export function last<T>(arr: readonly T[]): T | undefined {
+  if (!Array.isArray(arr)) {
+    throw new Error('not an array');
+  }
```
Comment (raon0211) [inline src/array/last.ts:28]:
```suggestion
```

I guess `arr` is guaranteed to be an array by types, so we might skip this check.

File: src/array/last.spec.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+      [3, 2],
+      [3, 3],
+    ];
+    expect(last(nestedArray)).toEqual([3, 3]);
+  });
+
+  it('throws an error if argument is not an array', () => {
+    // @ts-expect-error: Testing invalid input
+    expect(() => last('not an array')).toThrow();
+  });
```
Comment (raon0211) [inline src/array/last.spec.ts:35]:
```suggestion
```

File: docs/reference/array/last.md [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+- `arr`(`T[]`): The array from which to get the last element.
+
+### Returns
+
+(`T | undefined`): The last element of the array, or `undefined` if the array is empty.
+
+### Throws
+
+Throws an error if `arr` is not an array.
```
Comment (raon0211) [inline docs/reference/array/last.md:24]:
```suggestion
```

File: docs/reference/array/last.md [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+
+Returns the last element of an array.
+
+This function takes an array and returns the last element of the array. If the array is empty, the function returns `undefined`.
+
+## Signature
+
+```typescript
+export function last<T>(arr: readonly [...T[], T]): T;
+export function last<T>(arr: readonly T[]): T | undefined;
```
Comment (raon0211) [inline docs/reference/array/last.md:11]:
```suggestion
function last<T>(arr: readonly [...T[], T]): T;
function last<T>(arr: readonly T[]): T | undefined;
```

**Review by raon0211 — APPROVED** [review]
Looks great. Thanks!

**Pull request merged**